File: name-inverter/NameInverter/NameInverter/src/main/java/pl/frognar/nameinverter/NameInverter.java
package pl.frognar.nameinverter;

import java.util.List;

public class NameInverter {
    public static String invert(String name) {
        List<String> nameParts = List.of(name.trim().split(RegularExpressions.anyAmountOfWhitespace()));

        if (nameParts.size() < 2)
            return nameParts.get(0);

        return invert(nameParts);
    }

    private static String invert(List<String> nameParts) {
        String first = nameParts.get(0);
        String last = nameParts.get(1);

        return String.format("%s, %s", last, first);
    }
}

class RegularExpressions {
    public static String anyAmountOfWhitespace() {
        return "\\s+";
    }
}

File: name-inverter/name-inverter-py/05/NameInverter.py
import re
from typing import List


class NameInverter:
    @classmethod
    def invert(cls, name: str) -> str:
        name_parts = cls.break_into_parts(name)
        if len(name_parts) < 2:
            return name_parts[0]
        return cls.invert_parts(name_parts)

    @staticmethod
    def break_into_parts(name: str) -> List[str]:
        return re.split(RegularExpressions.any_amount_of_whitespaces(), name.strip())

    @classmethod
    def invert_parts(cls, name_parts: List[str]) -> str:
        first = name_parts[0]
        last = name_parts[1]
        return f'{last}, {first}'


class RegularExpressions:
    @staticmethod
    def any_amount_of_whitespaces() -> str:
        return '\\s+'
